### ISSUE TITLE
PM2.5 and PM10 caping at 150 ug/m3 

### ISSUE THREAD

**Issue by PetitSphincter** [libdyson-wg/ha-dyson#131]
Using a Dyson Purifier Humidity+Cool with this integration. Works like a charm. But during the 2 years i've used this integration i noticed something. 

The value of PM2.5 and PM10 are caped at 150 ug/m3 while the dyson cloud app or the display screen can show values up to 999 ug/m3. I just reproduced it right now by vaping right into the filter : 

![image](https://github.com/libdyson-wg/ha-dyson/assets/60610165/3bde664c-18c4-431a-a22d-ca4c959f1bbb)

While the value displayed on the screen was caping at 999 ug/m3 

All the value between 999 ug/m3 and 150 ug/m3 on the screen where at 150 ug/m3 on HA. Once it decreased under 150 ug/m3 values are the same, so it's not a problem of proportion just caping for values above 150.

**Comment by dotvezz** [comment]
Interesting! I'll look into this. For the time being would it also be possible to get a screenshot from your Dyson app showing the 999 value? I'm curious how it registers, and it could help us in understanding the signals.

**Comment by PetitSphincter** [comment]
Downloaded the app back (trying to get rid of cloud apps :) ) and destroyed my lungs but here's a screenshot : 

![Screenshot_20240221-193551](https://github.com/libdyson-wg/ha-dyson/assets/60610165/825d0bf7-fcd7-4ac0-9374-93af70b0cfb2)

And here's how it shows in HA : 

![image](https://github.com/libdyson-wg/ha-dyson/assets/60610165/4ba0bf84-7a05-4c8a-b672-e0f98c8e3fb4)

**Comment by PetitSphincter** [comment]
I'm not home currently coming home next wednesday, i just put a reminder for next week to try this out

**Comment by PetitSphincter** [comment]
This tend to raise VOC more than PM. Plus it's really hard to reach high level of PM with it. At least with air freshener i have

**Comment by PetitSphincter** [comment]
Struggled a bit to find the IP but after an angry IP scanner here's the results : 

![image](https://github.com/libdyson-wg/ha-dyson/assets/60610165/fd842e11-bcb9-4a32-97de-1b8534138a64)

It seems like the same things is happening here. The third scan is the one where it was way above the 150 ug/m3 (700 ug/m3). But here this point is at 0149 which is, i suppose, 149 ug/m3. All the other line where below on app, and coherent with the tool

**Comment by dotvezz** [comment]
@PetitSphincter thanks a ton for that info. It's really interesting, and kinda frustrating. Some devices to have values they send over MQTT which are flagged as "high resolution" in the mobile app. I'll investigate those values and their availability. If I'm able to work out a solution, I'll get back to you here!

**Comment by RedEarth1** [comment]
I've noticed this too, so I monitored my fan using MQTT explorer while giving the fan a quick squirt of deodorant and here's one of the messages:

`{"msg":"ENVIRONMENTAL-CURRENT-SENSOR-DATA","time":"2024-05-01T18:12:07.000Z","data":{"tact":"2963","hact":"0054","pm25":"0149","pm10":"0149","va10":"0013","noxl":"0007","p25r":"0999","p10r":"0999","hcho":"NONE","hchr":"NONE","sltm":"OFF"}}`

pm25 and pm10 are the values which were showing in HA (both 149) but as you can see p25r and p10r as both showing 999 which is also what was showing in the Dyson app at the time, so maybe these are the 'uncapped' values that it uses??

**Comment by dotvezz** [comment]
@RedEarth1 Yep, p10r and p25r are the "High Resolution" values. Thanks for looking into it!

The only reason I don't just immediately switch this integration to using them is because I'm not sure about the availability of them on the different devices and generations.

**Comment by DennisOtto-dev** [comment]
Just bought the Purifier Humidify-Cool Formaldehyde two days ago and I am experiencing the same issue regarding the PM-readings.
Yesterday there was a spike of more than 270 PM2.5 (looking at the Dyson App), but the highest PM2.5 value in my HomeAssistant during this time was only 132.
The same with PM10: Dyson App shows more than 270, HomeAssistant shows 104.
Looks like all values are off..
HCHO: Dyson App shows 0.011 and HomeAssistant shows around 0.018
VOC: Dyson App shows 3, HomeAssistant shows 4.74
NO2 looks correct, but for now I don't have any spikes to compare.

**Comment by PetitSphincter** [comment]
> @RedEarth1 Yep, p10r and p25r are the "High Resolution" values. Thanks for looking into it!
> 
> The only reason I don't just immediately switch this integration to using them is because I'm not sure about the availability of them on the different devices and generations.

Do you need any help on this ?

**Comment by ttravisswan** [comment]
I have a Dyson HP04 and noticed some odd behavior with the PM2.5 and PM10 values also. Before I used the Dyson integration in Home Assistant, I used the Dyson integration in Homebridge and joined the Homebridge instance with Home Assistant to expose these sensors and fan controls in HA. Below are two screenshots - one showing the PM10 readings from the sensor created by this HA Dyson integration, and the next showing the PM10 readings from the sensor created by Homebridge.

Home Assistant Dyson integration:
![image](https://github.com/user-attachments/assets/25d62c81-d4e5-44d0-b6f4-6f4409aab9a3)

Homebridge Dyson integration:
![image](https://github.com/user-attachments/assets/b729db24-64f9-4058-9d3c-07f7f90b8192)

It clearly looks like the two graphs trend up and down together, but the values are different. From what I can see in the Dyson app, the values in the Dyson -> Homebridge -> Home Assistant sensor are correct. For example, right before I took those screenshots, the PM10 value in the Dyson app jumped up to 3 ug/m3, but the HA Dyson integration never showed a value higher than 2 ug/m3. Happy to help troubleshoot this issue with you guys, just let me know what I can do to help.

**Comment by dotvezz** [comment]
I've released [v1.5.0](https://github.com/libdyson-wg/ha-dyson/releases/tag/v1.5.0), which uses "high resolution" sensors for PM2.5 and PM10. I'm super curious to know if this helps with the issue here!

**Comment by PetitSphincter** [comment]
I just tested it before going to work. At first glance, it works like a charm. The value on the app is the same as the value in HA. Thanks for this! 

Will test a bit more this week, but I think it's all good
